tests/lib/rules/no-jquery-methods.js

```diff
@@ -0,0 +1,100 @@
+const rule = require('../../../lib/rules/no-jquery-methods');
+const MESSAGE = rule.meta.message;
+const BLACKLISTMETHOD = rule.meta.blacklist[0];
+const RuleTester = require('eslint').RuleTester;
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('no-jquery-methods', rule, {
+  valid: [
+    {
+      code: `
+        export default Ember.Component({
+          init() {
+            Ember.$().notBlacklistedMethod();
+          }
+        });`
+    },
+    {
+      code: `
+        export default Ember.Component({
+          init() {
+            this.$().notBlacklistedMethod();
+          }
+        });`
+    },
+    {
+      code: `
+        export default Ember.Component({
+          init() {
+            const myJQueryObj = this.$();
+            myJQueryObj[${BLACKLISTMETHOD}]();      
```

Comment:
To add a bit more color to that last comment. I assumed this test was being marked as valid to acknowledge that we aren't specifically trying to catch usage from variables. I say this because doing it well will be very difficult. Consider:

```js
function show() {
  const el = getElement();
  el.show();
}
```

Or even:

```js
Component.extend({
  init() {
    this.input = this.$('input');
  },
  toggleInput() {
    this.input.toggle();
  }
});
```

If we do try to catch any of these, we should be explicit about what we can and cannot reasonably detect so that consumers understand the behavior accurately.